urlparse4/cgurl.pyx

```diff
@@ -429,19 +466,18 @@ def _safe_ParseResult(parts, encoding='utf8', path_encoding='utf8'):
         to_native_str(netloc),
 
         # default encoding for path component SHOULD be UTF-8
-        quote(to_bytes(parts.path, path_encoding), _safe_chars),
-        quote(to_bytes(parts.params, path_encoding), _safe_chars),
+        to_unicode(parts.path, path_encoding),
+        to_unicode(parts.params, path_encoding),
 
         # encoding of query and fragment follows page encoding
         # or form-charset (if known and passed)
-        quote(to_bytes(parts.query, encoding), _safe_chars),
-        quote(to_bytes(parts.fragment, encoding), _safe_chars)
+        to_unicode(parts.query, encoding),
+        to_unicode(parts.fragment, encoding)
     )
 
 def canonicalize_url(url, keep_blank_values=True, keep_fragments=False,
                      encoding=None):
     r"""Canonicalize the given url by applying the following procedures:
-
```

Comment:
I think these blank lines were here for a reason (nicer rst formatting, better block separation), let's keep them :)

Reply:
Sure I will do that! Thank you!